k8s-clouddriver-plugin/src/main/kotlin/com/amazon/spinnaker/clouddriver/k8s/services/GitRepoCredentials.kt

```diff
@@ -0,0 +1,94 @@
+package com.amazon.spinnaker.clouddriver.k8s.services
+
+import com.amazon.spinnaker.clouddriver.k8s.configuration.PluginConfig
+import com.amazon.spinnaker.clouddriver.k8s.exceptions.ResourceNotFound
+import com.amazon.spinnaker.clouddriver.k8s.model.GitRepo
+import com.netflix.spinnaker.clouddriver.artifacts.ArtifactCredentialsRepository
+import com.netflix.spinnaker.clouddriver.artifacts.gitRepo.GitJobExecutor
+import com.netflix.spinnaker.clouddriver.artifacts.gitRepo.GitRepoArtifactCredentials
+import com.netflix.spinnaker.kork.exceptions.MissingCredentialsException
+import org.slf4j.LoggerFactory
+import java.io.File
+
+class GitRepoCredentials(
+    private val artifactCredentialsRepository: ArtifactCredentialsRepository,
+    private val pluginConfig: PluginConfig
+) {
+
+    private val executorField = GitRepoArtifactCredentials::class.java.getDeclaredField("executor")
+    private val logger = LoggerFactory.getLogger(GitRepoCredentials::class.java)
+
+    init {
+        executorField.isAccessible = true
+    }
+
+    fun getAllCredentials(): List<GitRepo> {
+        val creds = artifactCredentialsRepository.allCredentials
+        return creds.mapNotNull {
+            if (it !is GitRepoArtifactCredentials) {
+                null
+            } else {
+                val executor = executorField.get(it) as GitJobExecutor
+                val account = executor.account
+                val repo = GitRepo(
+                    account.name,
+                    account.token,
+                    account.username,
+                    account.password,
+                    repos = getReposForAccount(account.name)
+                )
+                if (!account.sshPrivateKeyFilePath.isNullOrEmpty()) {
+                    repo.sshPrivateKey = readSSHKey(account.sshPrivateKeyFilePath)
+                }
+                return@mapNotNull repo
+            }
+        }
+    }
+
+    fun getCredentials(name: String): GitRepo {
+        try {
+            val cred = artifactCredentialsRepository.getCredentials(name, "git/repo")
+            cred.let {
+                val executor = executorField.get(it) as GitJobExecutor
+                val account = executor.account
+                val out = GitRepo(
+                    name,
+                    account.token,
+                    account.username,
+                    account.password,
+                    repos = getReposForAccount(name),
+                )
+                if (!account.sshPrivateKeyFilePath.isNullOrEmpty()) {
+                    out.sshPrivateKey = readSSHKey(account.sshPrivateKeyFilePath)
+                }
+                return out
+            }
+        } catch (e: MissingCredentialsException) {
+            logger.info("credential with name, {}, not found", name)
+            throw ResourceNotFound()
+        }
+    }
+
+    // SSH Key files should be less than 16KB in size with 16384 bits. ECC keys should be much smaller.
```

Comment:
```suggestion
    // SSH Key files should be less than 16KB in size with 16384 bytes. ECC keys should be much smaller.
```

Reply:
I should be more clear there actually. 16384 bits is the max value you can specify to be in the key when generating ssh key pairs with ssh-keygen. Resulting private key file should be well below 16KB.